module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -398,11 +398,15 @@ zfs_zrele_async(znode_t *zp)
 	ASSERT(atomic_read(&ip->i_count) > 0);
 	ASSERT(os != NULL);
 
-	if (atomic_read(&ip->i_count) == 1)
+	/*
+	 * If decrementing the count would put us at 0, we can't do it inline
+	 * here, because that would be synchronous. Instead, dispatch an iput
```

Comment:
Could you talk about why being synchronous would be bad?  e.g. the code path that it can call back into zfs and deadlock

Reply:
There's a pretty good description of when `zrele_async` should be used in [zfs_vnops_os.c](https://github.com/openzfs/zfs/blob/master/module/os/linux/zfs/zfs_vnops_os.c#L90-L98).  Perhaps you could reference that explanation here and expand on it a little bit.